### PR TITLE
feat: add password hashing

### LINKED ISSUE #5: Add password Hashing in signup and signin
Hash user's password before sending request to db 

similarly compare the users given password with the hash from the db and if hash matches the login else popup wrong password

### PR DESCRIPTION
**Pull request by pandeyashu08**
This pr setups: 
- [x] Password Hashing Using bcrypt.js.
- [x] Password Comparison in Sign in handler. 
- [x] Refactored Frontend Logic for Sign in.

### Before: 
Screenshot 2024-10-14 001226

### After:
Screenshot 2024-10-14 001242

### CODE REVIEW

**Comment by pandeyashu08** [comment]
@yesahem @geekyharsh05  Can you please review it?

**Comment by geekyharsh05** [comment]
@pandeyashu08 LGTM!

File: backend/src/controllers/user.controller.ts [hunk 1]
```diff
@@ -7,6 +7,9 @@ import {
   SigninSchema,
 } from "@shishuranjan/backend-common/dist/validations";
 import { generateToken } from "../utils/generateToken";
+import bcrypt from 'bcryptjs'
+import { log } from "console";
+import { get } from "http";
```
Comment (geekyharsh05) [inline backend/src/controllers/user.controller.ts]: What's up with this sir? Remove these.

**Review by geekyharsh05 — CHANGES_REQUESTED** [review]

File: backend/src/controllers/user.controller.ts (hunk 1 above)
Comment (pandeyashu08) [reply backend/src/controllers/user.controller.ts]: sorry i forgot to remove unused imports , i will do it on the fly

Comment (yesahem) [reply backend/src/controllers/user.controller.ts]: do it asap @pandeyashu08 Its looks good for merging !!!

**Comment by yesahem** [comment]
Looks Cool @pandeyashu08 Thanks for contribution 
Merging this @geekyharsh05

**Pull request merged**